orte/mca/rml/ofi/rml_ofi_component.c

```diff
@@ -0,0 +1,1091 @@
+                                        "\n Protocol  : %d", *protocol_ptr);  //fi_tostr(protocol_ptr,FI_TYPE_PROTOCOL));
+                }
+                if( orte_get_attribute( prov, ORTE_PROV_NAME, (void **)&prov_name ,OPAL_STRING)) {
+                    opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                                        "\n Provider name : %s",prov_name);
+                } else {
+                    opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                                    "\n Error in getting Provider name");
+                }
+            } else {
+                opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                                    "\n %s:%d opal_value_unload() failed, opal_list* prov = %x",
+                                   __FILE__,__LINE__,prov);
+            }
+            providers = (opal_value_t *)providers->super.opal_list_next;
+            /*  opal_output_verbose(1,orte_rml_base_framework.framework_output,"\n %s:%d 
+             * - Moving on to next provider provders=%x",__FILE__,__LINE__,providers);  
+            */
+        }
+    } else {
+        opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                            "\n query_transports() returned Error ");
+    }
+    opal_output_verbose(10,orte_rml_base_framework.framework_output,
+                        "\n End of print_transports_query() \n");
+}
+
+
+
+
+/**
+    conduit [in]: the conduit_id that triggered the progress fn
+ **/
+__opal_attribute_always_inline__ static inline int
+orte_rml_ofi_progress(ofi_transport_conduit_t* conduit)
+{
+    ssize_t ret;
+    int count=0;    /* number of messages read and processed */
+    struct fi_cq_data_entry wc = { 0 };
+    struct fi_cq_err_entry error = { 0 };
+    orte_rml_ofi_request_t *ofi_req;
+
+    opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                         "%s orte_rml_ofi_progress called for conduitid %d",
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),
+                         conduit->conduit_id);
+    /**
+    * Read the work completions from the CQ.
+    * From the completion's op_context, we get the associated OFI request.
+    * Call the request's callback.
+    */
+    while (true) {
+        /* Read the cq - that triggered the libevent to call this progress fn. */
+        ret = fi_cq_read(conduit->cq, (void *)&wc, 1);
+        if (0 < ret) {
+            opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                         "%s cq read for conduitid %d - wc.flags = %x",
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),
+                         conduit->conduit_id, wc.flags);
+            count++;
+            // check the flags to see if this is a send-completion or receive
+            if ( (wc.flags & FI_SEND) || ((wc.flags & FI_SEND) && (wc.flags & FI_TRANSMIT_COMPLETE)) )
```

Comment:
The 2nd half of this if statement looks redundant -- the statement will always be true if it's FI_SEND, regardless of whether FI_TRANSMIT_COMPLETE is set or not.